File: back/DevC/src/main/java/com/back/devc/domain/interaction/report/dto/ReportResponseDTO.java
package com.back.devc.domain.interaction.report.dto;

import com.back.devc.domain.interaction.report.entity.Report;
import com.back.devc.domain.interaction.report.entity.ReportStatus;
import com.back.devc.domain.interaction.report.entity.TargetType;

import java.time.LocalDateTime;

public record ReportResponseDTO(

        Long reportId,

        String reporterEmail,
        String reporterNickname,

        TargetType targetType,
        Long targetId,

        String targetNickname,
        String targetTitle,
        String targetContent,

        String reasonType,
        String reasonDetail,

        ReportStatus status,

        LocalDateTime createdAt,
        LocalDateTime processedAt
) {

    public static ReportResponseDTO of(
            Report report,
            String targetNickname,
            String targetTitle,
            String targetContent
    ) {
        return new ReportResponseDTO(
                report.getReportId(),
                report.getReporter().getEmail(),
                report.getReporter().getNickname(),
                report.getTargetType(),
                report.getTargetId(),
                targetNickname,
                targetTitle,
                targetContent,
                report.getReasonType(),
                report.getReasonDetail(),
                report.getStatus(),
                report.getCreatedAt(),
                report.getProcessedAt()
        );
    }
}

File: back/DevC/src/main/kotlin/com/back/devc/domain/interaction/report/dto/ReportResponseDTO.kt
package com.back.devc.domain.interaction.report.dto

import com.back.devc.domain.interaction.report.entity.Report
import com.back.devc.domain.interaction.report.entity.ReportStatus
import com.back.devc.domain.interaction.report.entity.TargetType
import java.time.LocalDateTime

data class ReportResponseDTO(
    val reportId: Long,

    val reporterEmail: String,

    val reporterNickname: String,

    val targetType: TargetType,

    val targetId: Long,

    val targetNickname: String?,

    val targetTitle: String?,

    val targetContent: String?,

    val reasonType: String,

    val reasonDetail: String?,

    val status: ReportStatus,

    val createdAt: LocalDateTime,

    val processedAt: LocalDateTime?
) {
    companion object {
        fun of(
            report: Report,
            targetNickname: String?,
            targetTitle: String?,
            targetContent: String?
        ): ReportResponseDTO {
            val reporter = report.reporter

            return ReportResponseDTO(
                reportId = report.reportId
                    ?: throw IllegalStateException("Persisted report id is required for response mapping"),
                reporterEmail = reporter.email,
                reporterNickname = reporter.nickname,
                targetType = report.targetType,
                targetId = report.targetId,
                targetNickname = targetNickname,
                targetTitle = targetTitle,
                targetContent = targetContent,
                reasonType = report.reasonType,
                reasonDetail = report.reasonDetail,
                status = report.status,
                createdAt = report.createdAt
                    ?: throw IllegalStateException("Report creation time is required for response mapping"),
                processedAt = report.processedAt
            )
        }
    }
}
